labs-ios-starter/Map/LocationSearchTableViewController.swift

```diff
@@ -47,38 +51,57 @@ class LocationSearchTableViewController: UITableViewController {
     // MARK: - Table view data source
 
     override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        return matchingItems.count
+        return citySearchResults.count
     }
     
     override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
         let cell = tableView.dequeueReusableCell(withIdentifier: "cell")!
-        let selectedItem = matchingItems[indexPath.row].placemark
-        cell.textLabel?.text = selectedItem.name
-        cell.detailTextLabel?.text = parseAddress(selectedItem: selectedItem)
+        //let selectedItem = matchingItems[indexPath.row].placemark
+        //cell.textLabel?.text = selectedItem.name
+        //cell.detailTextLabel?.text = parseAddress(selectedItem: selectedItem)
+        let city: String = citySearchResults[indexPath.row]
+        let cityName: String = String(city.dropLast(3))
+        let state: String = String(city.dropFirst(city.count-2))
+        cell.textLabel?.text = cityName
+        cell.detailTextLabel?.text = state
         return cell 
     }
     
+    /*
     override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
         let selectedItem = matchingItems[indexPath.row].placemark
         handleMapSearchDelegate?.dropPinZoomIn(placemark: selectedItem)
         dismiss(animated: true, completion: nil)
-    }
+    }*/
 
 }
 
 extension LocationSearchTableViewController: UISearchResultsUpdating {
+    
+    
     func updateSearchResults(for searchController: UISearchController) {
         guard let mapView = mapView,
               let searchBarText = searchController.searchBar.text else { return }
+        
+        let filteredCities = cities.filter { (city) -> Bool in
+            return city.starts(with: searchBarText)
+        }
+        self.citySearchResults = filteredCities
+        self.tableView.reloadData()
+        
+        /*
         let request = MKLocalSearch.Request()
         request.naturalLanguageQuery = searchBarText
         request.region = mapView.region
+        request.resultTypes = .address
+        
         let search = MKLocalSearch(request: request)
         search.start { response, _ in
             guard let response = response else { return }
             self.matchingItems = response.mapItems
+            
             self.tableView.reloadData()
```

Comment:
When this gets called is it parsing the file again? Since it's in the View Did load function?